toplevel/role/role.go

```diff
@@ -221,6 +228,61 @@ func (c config) Apply(address string, entriesBytes []byte, dryRun bool, threadPo
 	return nil
 }
 
+// Extracts names of policies referenced within applicable properties of desired roles
+// and updates those properties to only include the names of policies.
+// This is necessary to support policy file references within schemas.
+// The graphql query will return nested object for policy properties but vault api
+// expects only names of policies to be included
+func formatPolicyRefs(desiredRoles []entry) error {
+	// defines applicable properties(s) for each role type to reformat
+	// see https://github.com/app-sre/qontract-schemas/blob/main/schemas/vault-config/role-1.yml
+	typeToProperties := map[string][]string{
+		"approle": {
+			"token_policies",
+			"policies",
+		},
+		"oidc": {
+			"token_policies",
+		},
+		"kubernetes": {
+			"token_policies",
+		},
+	}
+	extracted := []string{}
+	for _, role := range desiredRoles {
+		for _, property := range typeToProperties[role.Type] {
+			policies, ok := role.Options[property].([]interface{})
+			if !ok {
+				return fmt.Errorf("[Vault Role] Failed to convert policy list `%s` within `%s`",
```

Comment:
I recommend removing the `[Vault Role]` tag from this error, as this is a utility function, and improving it ever so slightly so that it is more self-contained. 

This would align more with Go's concept of wrapping errors, where their methods in the chain keep wrapping errors until some top-level caller unwraps and prints a more concrete error message.

For a long line, this would be fine, though.

The only other place where we do it would be:

![Screenshot from 2023-08-17 14-20-47](https://github.com/app-sre/vault-manager/assets/250723/a1bcf514-dafc-4a03-a369-7654e852f8f0)

Reply:
Great point. Will push update. The whole error message structure in this app is probably due for a refactor at some point.